Scripts/Helpers/ProxyNode.cs

```diff
@@ -0,0 +1,619 @@
+                        continue;
+
+                    if (request == null || feedbackRequest.priority >= request.priority)
+                        request = feedbackRequest;
+                }
+
+                if (request == null)
+                    return;
+
+                affordanceData.SetVisible(true, request.duration);
+
+                foreach (var renderer in affordanceData.renderers)
+                {
+                    if (renderer)
+                        this.SetHighlight(renderer.gameObject, !request.suppressExisting);
+                }
+
+                var tooltipText = request.tooltipText;
+                if (!string.IsNullOrEmpty(tooltipText) || request.suppressExisting)
+                {
+                    foreach (var tooltip in affordanceData.tooltips)
+                    {
+                        if (tooltip)
+                        {
+                            tooltip.tooltipText = tooltipText;
+                            this.ShowTooltip(tooltip, true, placement: tooltip.GetPlacement(m_FacingDirection));
+                        }
+                    }
+                }
+            }
+        }
+
+        public void RemoveFeedbackRequest(ProxyFeedbackRequest request)
+        {
+            AffordanceData affordanceData;
+            if (m_AffordanceData.TryGetValue(request.control, out affordanceData))
+            {
+                foreach (var renderer in affordanceData.renderers)
+                {
+                    if (renderer)
+                        this.SetHighlight(renderer.gameObject, false);
+                }
+
+                affordanceData.SetVisible(false);
+
+                if (!string.IsNullOrEmpty(request.tooltipText))
+                {
+                    foreach (var tooltip in affordanceData.tooltips)
+                    {
+                        if (tooltip)
+                        {
+                            tooltip.tooltipText = string.Empty;
+                            this.HideTooltip(tooltip, true);
+                        }
+                    }
+                }
+            }
+
+            foreach (var feedbackRequest in m_FeedbackRequests)
+            {
+                if (feedbackRequest == request)
+                {
+                    m_FeedbackRequests.Remove(feedbackRequest);
+                    if (!request.showBody)
+                        ExecuteFeedback(request);
+
+                    break;
+                }
+            }
+        }
+
+        public void ClearFeedbackRequests(IRequestFeedback caller)
+        {
+            k_FeedbackRequestsCopy.Clear();
+            foreach (var request in m_FeedbackRequests)
+            {
+                if (request.caller == caller)
+                    k_FeedbackRequestsCopy.Add(request);
+            }
+
+            foreach (var request in k_FeedbackRequestsCopy)
+            {
+                RemoveFeedbackRequest(request);
+            }
```

Comment:
you might want another clear of k_FeedbackRequestsCopy here - which I know is redundant but for GC purposes it just makes sure certain assets aren't referenced